From eeab628f758b108aee042d7cadf17105e47f4825 Mon Sep 17 00:00:00 2001
From: Jai Luthra <j-luthra@ti.com>
Date: Fri, 31 May 2024 14:01:09 +0530
Subject: [PATCH 300/465] media: i2c: imx390: Add 100Khz input clock margin

While the sensor only supports input clock from 6-27Mhz, some clock
sources might not have exact PLL dividers.

Adding 100Khz margin allows poor resolution dividers to still hit ~27Mhz
and run the sensor at full framerate.

Signed-off-by: Jai Luthra <j-luthra@ti.com>
---
 drivers/media/i2c/imx390.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/i2c/imx390.c b/drivers/media/i2c/imx390.c
index 2439d5e1533c..09ccfebcd98c 100644
--- a/drivers/media/i2c/imx390.c
+++ b/drivers/media/i2c/imx390.c
@@ -643,7 +643,7 @@ static int imx390_probe(struct i2c_client *client)
 	imx390->clk_rate = clk_get_rate(imx390->clk);
 	dev_info(imx390->dev, "inck rate: %lu Hz\n", imx390->clk_rate);
 
-	if (imx390->clk_rate < 6000000 || imx390->clk_rate > 27000000)
+	if (imx390->clk_rate < 5900000 || imx390->clk_rate > 27100000)
 		return -EINVAL;
 
 	ret = imx390_power_on(imx390);
-- 
2.45.1

